modules/lib/lib-scheduler/src/main/resources/lib/xp/scheduler.ts

```diff
@@ -21,54 +21,47 @@ function checkRequired<T extends object>(obj: T, name: keyof T): void {
 
 export type EditorFn<T> = (value: T) => T;
 
-export type ScheduleType = 'CRON' | 'ONE_TIME';
-
-export interface Schedule {
-    value: string;
-    type: ScheduleType;
-    timezone: string;
-}
-
-export interface CreateScheduledJobParams {
+export interface CreateScheduledJobParams<Config extends object = Record<string, unknown>> {
     name: string;
-    description: string;
+    description?: string;
     descriptor: string;
-    config: Record<string, unknown>;
-    schedule: Schedule;
-    user: string;
+    config?: Config;
+    schedule: OneTimeSchedule | CronSchedule;
+    user?: string;
     enabled: boolean;
 }
 
-export interface OnTimeSchedule {
+export interface OneTimeSchedule {
+    type: 'ONE_TIME';
     value: string;
-    type: ScheduleType;
 }
 
-export interface CronSchedule
-    extends OnTimeSchedule {
+export interface CronSchedule {
+    type: 'CRON';
+    value: string;
     timeZone: string;
 }
 
-export interface ScheduleJob {
+export interface ScheduledJob<Config extends object = Record<string, unknown>> {
     name: string;
     descriptor: string;
-    description: string;
+    description?: string | null;
     enabled: boolean;
-    config?: Record<string, unknown> | null;
+    config?: Config | null;
     user?: string | null;
-    creator?: string | null;
-    modifier?: string | null;
-    createdTime?: string | null;
-    modifiedTime?: string | null;
+    creator: string;
+    modifier: string;
+    createdTime: string;
+    modifiedTime: string;
     lastRun?: string | null;
     lastTaskId?: string | null;
-    schedule: OnTimeSchedule | CronSchedule;
+    schedule: OneTimeSchedule | CronSchedule;
 }
 
-interface CreateScheduledJobHandler {
+interface CreateScheduledJobHandler<Config extends object = Record<string, unknown>> {
```

Comment:
I would probably not set av default type parameter for Handlers (like `CreateScheduledJobHandler` or `ModifyScheduledJobParams` or `GetScheduledJobHandler` etc,).

These interface are not exposed externally, and it's better if we just always pass on the `<Config>` shape from the calling function.

```suggestion
interface CreateScheduledJobHandler<Config extends object> {
```